dm_control/locomotion/walkers/assets/dog_v2/utils.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2023 The dm_control Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or  implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+
+import os
+
+from dm_control.mujoco.wrapper import util
+from dm_control.suite import common
+from dm_control.utils import io as resources
+from dm_control.utils import xml_tools
+
+import numpy as np
+
+from scipy.spatial.transform import Rotation as Rot
+
+
+from muscles import lateral, flexors_back, flexors_front, \
+    extensors_back, extensors_front
+
+muscle_legs = extensors_back + extensors_front + \
+    flexors_back + flexors_front
+
+
+def array_to_string(array):
+    return ' '.join(['%8g' % num for num in array])
+
+
+def calculate_transformation(element):
```

Comment:
Please add docstrings to this and all other nontrivial functions in this file

Reply:
+1